src/Options/DataAnnotations/src/DataAnnotationValidateOptions.cs

```diff
@@ -48,10 +48,12 @@ public ValidateOptionsResult Validate(string name, TOptions options)
                     return ValidateOptionsResult.Success;
                 }
 
-                return ValidateOptionsResult.Fail(String.Join(Environment.NewLine,
-                    validationResults.Select(r => "DataAnnotation validation failed for members " +
-                        String.Join(", ", r.MemberNames) +
-                        " with the error '" + r.ErrorMessage + "'.")));
+                var errors = new List<string>();
+                foreach (var r in validationResults)
+                {
+                    errors.Add($"DataAnnotation validation failed for members: '{String.Join(",", r.MemberNames)}' with the error: '{r.ErrorMessage}'.");
```

Comment:
@blowdart Is it okay to have actual validation failure messages in exception text? Might not these messages have sensitive data?

Reply:
Ugh. No way to know.

Reply:
Agreed. So can they go in exception text or not?

Reply:
Its pretty useless if we don't show the actual validation error, that's kind of the whole point for this change, to put the validation error in their face...

Reply:
Otherwise we might as well just keep the old behavior which is you get an OptionsValidationException with the Failures inside and hard to discover

Reply:
Although I guess for data annotations specifically, we were already leaking the exception message, so maybe we need to resolve that either way

Reply:
Agreed it is unfriendly. Totally favor of this changes if @blowdart says its okay.

Reply:
PG decision, is the risk worth the rewards? Arthur seems to think yes.

Reply:
If the advise is anything other than, "no", then I say "yes!". But I'm going to make sure I keep track that you gave an implicit yes.